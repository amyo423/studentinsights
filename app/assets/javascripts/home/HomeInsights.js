import React from 'react';
import Card from '../components/Card';
import CheckStudentsWithLowGrades from './CheckStudentsWithLowGrades';


// On the home page, show users the answers to their most
// important questions.
class HomeInsights extends React.Component {
  render() {
    return (
      <div className="HomeInsights" style={styles.root}>
        <CheckStudentsWithLowGrades />
        {this.renderPlaceholder()}
      </div>
    );
  }

  renderPlaceholder() {
    return (
      <Card style={styles.card}>
        <div>
          <div>What else would help you support your students?</div>
          <div>Come talk with us about what we should build next!</div>
        </div>
      </Card>
    );
  }
}
HomeInsights.propTypes = {
  educatorId: React.PropTypes.number.isRequired
};

const styles = {
  root: {
    fontSize: 14
  },
  card: {
    margin: 10,
    marginTop: 20,
    border: '1px solid #ccc',
    borderRadius: 3
  }
};


export default HomeInsights;