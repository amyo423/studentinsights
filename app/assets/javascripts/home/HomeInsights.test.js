import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import fetchMock from 'fetch-mock/es5/client';
import HomeInsights, {UnsupportedStudentsPure} from './HomeInsights';
import SpecSugar from '../../../../spec/javascripts/support/spec_sugar.jsx';
import unsupportedLowGradesJson from '../../../../spec/javascripts/fixtures/home_unsupported_low_grades_json';


beforeEach(() => {
  fetchMock.restore();
  fetchMock.get('/home/unsupported_low_grades_json?limit=100', unsupportedLowGradesJson);
});

it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(<HomeInsights />, el);
});

SpecSugar.withTestEl('integration tests', container => {
  it('renders everything after fetch', done => {
    const el = container.testEl;
    ReactDOM.render(<HomeInsights />, el);
    
    setTimeout(() => {
      expect(el.innerHTML).toContain('Students to check on');
      done();
    }, 0);
  });
});

it('pure component matches snapshot', () => {
  const props = {
    limit: unsupportedLowGradesJson.limit,
    totalCount: unsupportedLowGradesJson.total_count,
    assignments: unsupportedLowGradesJson.assignments
  };
  const tree = renderer
    .create(<UnsupportedStudentsPure {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('handles when limit reached', () => {
  const props = {
    limit: 3,
    totalCount: 129,
    assignments: unsupportedLowGradesJson.assignments.slice(0, 3)
  };
  const tree = renderer
    .create(<UnsupportedStudentsPure {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
