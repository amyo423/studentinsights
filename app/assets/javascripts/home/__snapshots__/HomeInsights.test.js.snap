// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles when limit reached 1`] = `
<div
  className="UnsupportedStudentsPure"
>
  <div
    style={
      Object {
        "backgroundColor": "#eee",
        "borderBottom": "1px solid #ccc",
        "color": "black",
        "padding": 10,
      }
    }
  >
    Students to check on
  </div>
  <div
    className="Card "
    style={
      Object {
        "border": "none",
        "borderRadius": 3,
        "padding": 10,
      }
    }
  >
    <div>
      There are 
      <b>
        29
         students
      </b>
       you work with who have a D or an F right now but haven't been mentioned in NGE or 10GE for the last month.
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <div
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <div>
          <span>
            <a
              href="/students/67"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Pocahontas
               
              Skywalker
            </a>
          </span>
          <span>
             has a 
            50
             in 
            <a
              href="/sections/1"
            >
              SHS-BIO-TUES
            </a>
          </span>
          <span>
             with 
            <a
              className="Educator"
              href="mailto:bill@demo.studentinsights.org"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Bill Teacher
            </a>
          </span>
        </div>
      </div>
      <div
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <div>
          <span>
            <a
              href="/students/115"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Minnie
               
              Pan
            </a>
          </span>
          <span>
             has a 
            50
             in 
            <a
              href="/sections/6"
            >
              SCI-201B
            </a>
          </span>
          <span>
             with 
            <a
              className="Educator"
              href="mailto:fatima@demo.studentinsights.org"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Fatima Teacher
            </a>
          </span>
        </div>
      </div>
      <div
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <div>
          <span>
            <a
              href="/students/73"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Pocahontas
               
              Kenobi
            </a>
          </span>
          <span>
             has a 
            52
             in 
            <a
              href="/sections/1"
            >
              SHS-BIO-TUES
            </a>
          </span>
          <span>
             with 
            <a
              className="Educator"
              href="mailto:bill@demo.studentinsights.org"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Bill Teacher
            </a>
          </span>
        </div>
      </div>
    </div>
    <div>
      There are 
      29
       students total.  Start with checking in on these first 
      3
       students.
    </div>
  </div>
</div>
`;

exports[`pure component matches snapshot 1`] = `
<div
  className="UnsupportedStudentsPure"
>
  <div
    style={
      Object {
        "backgroundColor": "#eee",
        "borderBottom": "1px solid #ccc",
        "color": "black",
        "padding": 10,
      }
    }
  >
    Students to check on
  </div>
  <div
    className="Card "
    style={
      Object {
        "border": "none",
        "borderRadius": 3,
        "padding": 10,
      }
    }
  >
    <div>
      There are 
      <b>
        14
         students
      </b>
       you work with who have a D or an F right now but haven't been mentioned in NGE or 10GE for the last month.
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <div
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <div>
          <span>
            <a
              href="/students/67"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Pocahontas
               
              Skywalker
            </a>
          </span>
          <span>
             has a 
            50
             in 
            <a
              href="/sections/1"
            >
              SHS-BIO-TUES
            </a>
          </span>
          <span>
             with 
            <a
              className="Educator"
              href="mailto:bill@demo.studentinsights.org"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Bill Teacher
            </a>
          </span>
        </div>
      </div>
      <div
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <div>
          <span>
            <a
              href="/students/115"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Minnie
               
              Pan
            </a>
          </span>
          <span>
             has a 
            50
             in 
            <a
              href="/sections/6"
            >
              SCI-201B
            </a>
          </span>
          <span>
             with 
            <a
              className="Educator"
              href="mailto:fatima@demo.studentinsights.org"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Fatima Teacher
            </a>
          </span>
        </div>
      </div>
      <div
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <div>
          <span>
            <a
              href="/students/73"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Pocahontas
               
              Kenobi
            </a>
          </span>
          <span>
             has a 
            52
             in 
            <a
              href="/sections/1"
            >
              SHS-BIO-TUES
            </a>
          </span>
          <span>
             with 
            <a
              className="Educator"
              href="mailto:bill@demo.studentinsights.org"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Bill Teacher
            </a>
          </span>
        </div>
      </div>
      <div
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <div>
          <span>
            <a
              href="/students/87"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Pocahontas
               
              Kenobi
            </a>
          </span>
          <span>
             has a 
            55
             in 
            <a
              href="/sections/3"
            >
              ART-302A
            </a>
          </span>
          <span>
             with 
            <a
              className="Educator"
              href="mailto:hugo@demo.studentinsights.org"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Hugo Teacher
            </a>
          </span>
        </div>
      </div>
    </div>
    <div>
      <a
        href="#"
        onClick={[Function]}
      >
        See more
      </a>
    </div>
  </div>
</div>
`;
